### PR TITLE
Fail Early

### PR DESCRIPTION
**Pull request by jibsonline**
1. backoff time was not being reset for each IP and Credential thus exponentially growing
2. Check if IP is reachable before going into the loop of IPs and Creds

### CODE REVIEW

**Comment by jibsonline** [comment]
@joelrebel

File: login.go [hunk 1]
```diff
@@ -73,20 +73,25 @@ func (p *Params) Login() (connection interface{}, loginInfo LoginInfo, err error
...
-	for _, credentials := range p.Credentials {
+	//for each IpAddress
+	for _, ip := range p.IpAddresses {
+		if ip == "" {
+			continue
+		}
 
-		//for each credential k, v
-		for user, pass := range credentials {
+		hostreachable, _ := p.IsHostReachable(ip, 443)
```
Comment (joelrebel) [inline login.go:83]: Is the error ignored here on purpose?

File: login.go [hunk 2]
```diff
@@ -73,20 +73,25 @@ func (p *Params) Login() (connection interface{}, loginInfo LoginInfo, err error
...
-			for _, ip := range p.IpAddresses {
-				if ip == "" {
-					continue
-				}
+		//for credential map in slice
+		for _, credentials := range p.Credentials {
 
+			//for each credential k, v
+			for user, pass := range credentials {
+				backoff = 0
```
Comment (joelrebel) [inline login.go:94]: As a suggestion for a future change, switch to https://pkg.go.dev/github.com/cenkalti/backoff/v4 for dealing with backoffs.

**Review by joelrebel — APPROVED** [review]
This was probably the worst Go I've written, thank you for fixing it 🚀 !

File: login.go (hunk 2 above)
Comment (jibsonline) [reply login.go:94]: Will keep in mind. Thanks @joelrebel

File: login.go (hunk 1 above)
Comment (jibsonline) [reply login.go:83]: Yes, since I have handled it with the flag

**Pull request merged**